soc/arm/st_stm32/stm32h5/soc.c

```diff
@@ -32,8 +32,10 @@ LOG_MODULE_REGISTER(soc);
 static int stm32h5_init(void)
 {
 	/* Enable instruction cache in 1-way (direct mapped cache) */
-	LL_ICACHE_SetMode(LL_ICACHE_1WAY);
-	LL_ICACHE_Enable();
+	if (IS_ENABLED(CONFIG_ICACHE)) {
```

Comment:
There are many places where it is going to be forcefully enabled:
* [hwinfo](https://github.com/zephyrproject-rtos/zephyr/blob/b05148d4857b4f13f9e82e987f056f1af18630bb/drivers/hwinfo/hwinfo_stm32.c#L34C1-L36)
* [temp](https://github.com/zephyrproject-rtos/zephyr/blob/b05148d4857b4f13f9e82e987f056f1af18630bb/drivers/sensor/stm32_temp/stm32_temp.c#L128-L130)
* [vref](https://github.com/zephyrproject-rtos/zephyr/blob/b05148d4857b4f13f9e82e987f056f1af18630bb/drivers/sensor/stm32_vref/stm32_vref.c#L110C1-L112)